src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -241,6 +241,32 @@ public void UploadBook(string storageKeyOfBookFolder, string pathToBloomBookDire
 			DeleteFileSystemInfo(new DirectoryInfo(wrapperPath));
 		}
 
+		private void RemoveUnwantedAudioFiles(string destDirName, bool excludeNarrationAudio)
+		{
+			string audioDir = Path.Combine(destDirName, "audio");
+			if (!Directory.Exists(audioDir))
+				return;
+
+			foreach (var audioFile in Directory.EnumerateFiles(audioDir))
+			{
+				var match = RegexFileNameRegex.Match(audioFile);
+				if (match.Success)
+				{
+					Guid dummy;
+					if (Guid.TryParse(match.Groups[2].Value, out dummy))
+					{
+						// Always remove all .wav files with guid file names. They are the original recordings.
+						if (excludeNarrationAudio || audioFile.EndsWith(".wav"))
+							RobustFile.Delete(audioFile);
+					}
+				}
+			}
+
+			// We always leave the music files to get uploaded.
```

Comment:
This function doesn't work properly if the name of the background music is itself a GUID.

I could see 2 ways to deal with this:
1) Just explicitly write down this scenario in the comments (probably with a TODO) and say it doesn't work yet.  Also, probably need to re-word "we ALWAYS leave the music files to get uploaded" to "we try to leave all the music files to get uploaded" or something like that.
2) In BookStorageTests.cs, it identifies all the narration audio and all the background music which is actually used.  It wouldn't be too hard to do something similar to identify only the narration audio which is theoretically useful.  I was writing a new function (which isn't checked in yet) in HTMLDom to find the nodes part of it...  And then we just need to select the value of the "id" attribute.
 		public static XmlNodeList SelectAudioSentenceElements(XmlElement element)
		{
			return element.SafeSelectNodes(".//span[contains(@class,'audio-sentence')]");
		}

Reply:
Oh, i see you explicitly mentioned that in a different function...